packages/ui-components/src/Button/Button.tsx

```diff
@@ -0,0 +1,48 @@
+import React, { FunctionComponent } from "react";
+import classnames from "classnames/bind";
+
+import objectToClassnames from "../utils/objectToClassnames";
+
+import styles from "./Button.module.scss";
+
+type ButtonIntent =
+  | "primary-success"
+  | "primary-danger"
+  | "secondary"
+  | "tertiary";
+type ButtonType = "button" | "submit" | "reset";
+type ButtonSize = "standard" | "small";
+
+interface ButtonProps {
```

Comment:
I had a conversation with Andrii about strictly typing the props of the Button component to allow for the _unhandled props_ to be used without specifying every possible prop of an HTML `<button>`. His point is that if we are using strict TypeScript we shouldn't allow untyped props to be passed to the the underlying button element. I see his point, and so the compromise we arrived at was to intersect the Component props with the underlying HTML element props. Something like,

```jsx
type OwnButtonProps = {
  intent?: ButtonIntent;
  type?: ButtonType;
  size?: ButtonSize;
};

type NativeButtonProps = Omit<
  React.ButtonHTMLAttributes<HTMLButtonElement>,
  keyof OwnButtonProps
>;

export type ButtonProps = NativeButtonProps & OwnButtonProps;
```

You will notice this also reduces the repetition of underlying props like `id`, `ariaLabel`, and `onClick`. Since we are also mirroring the `type` property of the underlying `ButtonHTMLAttributes` we could also remove our own `ButtonType` as well, thus boiling  the component down to the purely cosmetic expression that the component really is. Because we aren't really adding any functionality to an HTML button, we are just controlling what it's styles are.

Reply:
Great suggestion.  That solves that issue nicely without creating huge type definitions.

Reply:
Note: I ended up putting the type prop back in.  The tests that set it via "setProp" fail but I don't want to hold things up so I've added a TODO to remove it and factor the tests when I can circle back around.